Mediator-Error_Handling.md

```diff
@@ -1,63 +1,121 @@
+  - G2B - Fallback to G2
+  - G3B - Send `e.p.msg.unsupported` and say what version(s) its supported
+
+- **G4** - When an internal error or any unexpected error happens.
+  - G4A - ignore (no effect/output) 
+  - G4B - Send a problem report "e.p.error"
+
+- **G5** - If the message is tampered (got crypto errors).
+  - G5A - ignore (no effect/output) 
+  - G5B - (sync!) Send a problem report "e.p.crypto"
+
+- **G6** - If the DID method is not supported (`did.peer` in this case)
+  - G6A - ignore (no effect/output)
+  - G6B - (sync!) e.p.did 
+
+- **G7** - If the DID method is malformed.
+  - G7A - ignore (no effect/output)
+  - G7B - (sync!) e.p.did.malformed
+
+- **G8** - When a parsing error from the decrypted message.
+  - G8A - ignore (no effect/output)
+  - G8B - (sync!) Send a `e.p.msg` If the plaintext message is malformed 
+  - G8C - Send a `e.p.msg.<PIURI>` If the plaintext message is malformed or if parsing into a specific protocol's data model fails.
+
+
 - Receive a problem report (1w):
 - in case of Warnings Reply `w.p` -> log warnings and escalate to an error `e.p` on the reply
 - in case of Error `e.p` -> log error
 
-- Traceability of the MsgID of the Problem Report to the original error (2d) -> ATL-4147
-- [optional] Log - https://input-output.atlassian.net/browse/ATL-4147
-- escalate_to must be configurable (1d)
-- [optional] update the protocol with new tokens (2d)
+#### Mediator especific Scenarios:
+
+- **M1** - Receive a pickup message from a DID not enrolled.
+  - M1A - ignore (no effect/output) 
+  - M1B - Send a problem report "e.p.req.not_enroll"
+
+- **M2** - Connection to the BD is not working (at the moment of the call).
+  - G4
+  - M2A - ignore (no effect/output)
+  - M2B - Send a problem report "e.p.me.res.storage" (maybe mention the try again later)
+
+- **M3** - Business logic in BD is failing (got some error from the query).
+  - G4
+  - M3A - ignore (no effect/output)
+  - M3B - Send a problem report "e.p.me.res.storage"
+
+- **M4** - Get a Forward message to a DID that is not enrolled.
+  - M4A - ignore (no effect/output)
+  - M4B - Send a problem report "e.p.req.not_enroll" (the `next` DID is not enrolled)
+
+- **M5** - LiveModeChange https://didcomm.org/messagepickup/3.0/live-delivery-change
+  - Follow the protocol -> Upon receiving the `live_delivery_change` message with `live_delivery` set to `true``:
+    - M5A - If the connection support live delivery respond with a [`status`](https://didcomm.org/messagepickup/3.0/status) message.
+    - M5B - If the connection is incapable of live delivery send a problem report `e.m.live-mode-not-supported`
+
+
+
+### Considerations
+
+In the Atala PRISM Mediator, when an issue arises, we undertake the following steps:
+* Log the error.
+* Dispatch a problem report as outlined in the aforementioned table.
+* Record the problem report and error information in the database.
+Every one of these steps should be traceable back to the initial call via a XRequestID,
```

Comment:
Every one of these steps should be traceable back to the initial call via an XRequestID, and by the MsgID (the SHA-251 hash of the encrypted message).